cache.go

```diff
@@ -200,6 +205,10 @@ func (c *Cache[T]) set(key string, value T, duration time.Duration, track bool)
 	return item
 }
 
+func (c *Cache[T]) setnx(key string, value T, duration time.Duration, track bool) *Item[T] {
```

Comment:
You can inline this within the public `Setnx` function. The reason `set` isn't inlined is because it's called from a number of places.

```go
func (c *Cache[T]) Setnx(key string, value T, duration time.Duration) {
  return c.bucket(key).setnx(key, value, duration, false)
}
```

Reply:
done, inline for setnx.